### PR TITLE
CONTRIBUTING.md file for PR guidelines check list

### PR DESCRIPTION
**Pull request by abaumann**
comment on changes to the markdown as you see fit.
This would show up as a link as shown on this page:
https://github.com/blog/1184-contributing-guidelines

- [x] **Submitter**: Rebase to master. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [ ] **Submitter**: **Assign PR to Tech Lead**
- [ ] **Tech Lead**: does a once-over of the PR
- [ ] **Tech Lead**: Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others - **LR assigns back to submitter** for updates
- repeat as necessary:
  * Rest of team may comment on PR at will
  * Further commits. DO NOT SQUASH. **Reassign to LR**
  * Update documentation as needed
  * **LR assigns to submitter**
  * Re-rebase to master.
- [x] **LR**: sign off on PR, assign to submitter to finalize PR
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge
- [x] **Submitter**: Check configuration files in Jenkins in case they need changes
- [ ] **Submitter**: Verify swagger UI on dev server still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+Add the following to your main PR comment
+=========================
+```
+- [ ] Submitter: Rebase to master. DO NOT SQUASH
+- [ ] Submitter: Make sure Swagger is updated if API changes
+- [ ] Submitter: Make sure documentation for code is complete
+- [ ] Submitter: Remove any TODOs that should be TODOne or create a story for further TODOs
+- [ ] Submitter: Include the JIRA issue number in the PR description
```
Comment (abaumann) [inline CONTRIBUTING.md]: suggested change:  
  Submitter: Include the JIRA issue number in the PR description - if applicable

**Comment by helgridly** [comment]
- boldface the whos (submitter, LR, dv)
- boldface "[re]assign to X" text

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  * Further commits. DO NOT SQUASH. Reassign to LR
+  * Update documentation as needed
+  * Further review. Reassign to submitter
+  * Re-rebase to master.
+- [ ] LR: sign off on PR, assign to submitter to finalize PR
+- [ ] Submitter: Verify all tests go green, including CI tests
+- [ ] Submitter: Squash commits and merge
+- [ ] Submitter: Check configuration files in Jenkins in case they need changes
+- [ ] Submitter: Verify deployment worked on dev server
+- [ ] Submitter: Double check documentation
```
Comment (helgridly) [inline CONTRIBUTING.md]: Unclear what this means. Code comments? Confluence? Swagger?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+Add the following to your main PR comment
+=========================
+```
+- [ ] Submitter: Rebase to master. DO NOT SQUASH
+- [ ] Submitter: Make sure Swagger is updated if API changes
+- [ ] Submitter: Make sure documentation for code is complete
+- [ ] Submitter: Remove any TODOs that should be TODOne or create a story for further TODOs
```
Comment (helgridly) [inline CONTRIBUTING.md]: this is cute but i wondered what Tod One was

File: CONTRIBUTING.md (hunk 1 above)
Comment (helgridly) [inline CONTRIBUTING.md]: The current text implies that PRs should have a JIRA issue number, which we want to encourage anyway. I vote leave it as is

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  * Rest of team may comment on PR at will
+  * Further commits. DO NOT SQUASH. Reassign to LR
+  * Update documentation as needed
+  * Further review. Reassign to submitter
+  * Re-rebase to master.
+- [ ] LR: sign off on PR, assign to submitter to finalize PR
+- [ ] Submitter: Verify all tests go green, including CI tests
+- [ ] Submitter: Squash commits and merge
+- [ ] Submitter: Check configuration files in Jenkins in case they need changes
+- [ ] Submitter: Verify deployment worked on dev server
```
Comment (helgridly) [inline CONTRIBUTING.md]: What does this entail?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+- [ ] Submitter: Double check documentation
+- [ ] Submitter: Inform other teams of any API changes
```
Comment (helgridly) [inline CONTRIBUTING.md]: By email?

**Comment by abaumann** [comment]
- [ ] **Submitter**: Rebase to master. DO NOT SQUASH
- [ ] **Submitter**: Make sure Swagger is updated if API changes
- [ ] **Submitter**: Make sure documentation for code is complete
- [ ] **Submitter**: Remove any TODOs that should be TODOne or create a story for further TODOs
- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [ ] **Submitter**: Assign PR to dvoet
- [ ] **dvoet**: does a once-over of the PR
- [ ] **dvoet**: Anoint a lead reviewer (LR). **Assign PR to LR**
- [ ] **LR**: Initial review by LR and others - **LR assigns** back to submitter for updates
- repeat as necessary:
  * Rest of team may comment on PR at will
  * Further commits. DO NOT SQUASH. **Reassign to LR**
  * Update documentation as needed
  * Further review. **Reassign to submitter**
  * Re-rebase to master.
- [ ] **LR: sign off on PR, assign to submitter to finalize PR
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge
- [ ] **Submitter**: Check configuration files in Jenkins in case they need changes
- [ ] **Submitter**: Verify deployment worked on dev server
- [ ] **Submitter**: Double check all code comments
- [ ] **Submitter**: Inform other teams of any API changes
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

**Comment by helgridly** [comment]
Missing boldface on "Assign PR to dvoet", I'd also boldface the entirety of "LR assigns back to submitter" and change the "further review" one to "LR reassigns to submitter"

Also your "sign off on PR" line is busted.

**Comment by helgridly** [comment]
Useful for testing: http://dillinger.io/

**Comment by helgridly** [comment]
I still want to know what our process is if Doug is on vacation. As it stands, he's a bottleneck (like right now, for example!).

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
+Add the following to your main PR comment
+=========================
+```
+- [ ] **Submitter**: Rebase to master. DO NOT SQUASH
+- [ ] **Submitter**: Make sure Swagger is updated if API changes
+- [ ] **Submitter**: Make sure documentation for code is complete
+- [ ] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
+- [ ] **Submitter**: Include the JIRA issue number in the PR description
+- [ ] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
+- [ ] **Submitter**: **Assign PR to Tech Lead**
```
Comment (helgridly) [inline CONTRIBUTING.md]:
```
- [ ] **Submitter**: **Assign PR to** ![](http://i.imgur.com/9dLzbPd.png)
```

**Comment by helgridly** [comment]
:+1: squash + merge away

**Pull request merged**